src/script/sign.cpp

```diff
@@ -152,7 +180,7 @@ bool ProduceSignature(const SigningProvider& provider, const BaseSignatureCreato
         // the final scriptSig is the signatures from that
         // and then the serialized subscript:
         subscript = CScript(result[0].begin(), result[0].end());
-        solved = solved && SignStep(provider, creator, subscript, result, whichType, SigVersion::BASE) && whichType != TX_SCRIPTHASH;
+        solved = solved && SignStep(provider, creator, subscript, result, whichType, SigVersion::BASE, sigdata) && whichType != TX_SCRIPTHASH;
```

Comment:
If you'd introduce a `sigdata.scripts.emplace(CScriptID(subscript), subscript);` here (and similarly below for the `TX_WITNESS_V0_SCRIPTHASH` case), you get (I think) all functionality you need for operating on `SignatureData` objects without the need for the legacy partial signatures.

Reply:
Done